packages/overlay/src/sidecar.ts

```diff
@@ -8,7 +8,9 @@ export function connectToSidecar(
   setOnline: React.Dispatch<React.SetStateAction<boolean>>,
 ): () => void {
   log('Connecting to sidecar at', sidecarUrl);
-  const source = new EventSource(sidecarUrl);
+  const { origin } = new URL(sidecarUrl);
+  const sidecarStreamUrl: string = `${origin}/stream`;
+  const source = new EventSource(sidecarStreamUrl);
```

Comment:
Feels like this should be the ground truth for the sidecarURL and we should not have anything stored in `sentryDataCache` regarding the URL?

Reply:
Hey @BYK, could you please explain what you meant here?

The sidecarURL is either provided by the user or defaults to a pre-set URL, and then it's passed here in this file. Ideally, to use the sidecarURL effectively, we should store it somewhere.

Are you suggesting that instead of storing it in SentryDataCache, we should store it somewhere else, like in a context, so that other integrations can also access the sidecarURL?